### PR TITLE
CNV-43689: Marked primary pod network as optional in YAML spec

### PR DESCRIPTION
**Pull request by sjhala-ccs**
Version(s): 4.15+

Issue: [CNV-43689](https://issues.redhat.com//browse/CNV-43689)

Link to docs preview: 

- Linux bridge network: https://79224--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/vm_networking/virt-connecting-vm-to-linux-bridge.html#virt-attaching-vm-secondary-network-cli_virt-connecting-vm-to-linux-bridge
- SR-IOV network: https://79224--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/vm_networking/virt-connecting-vm-to-sriov.html#virt-attaching-vm-to-sriov-network_virt-connecting-vm-to-sriov
- OVN-Kubernetes network: https://79224--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/vm_networking/virt-connecting-vm-to-ovn-secondary-network.html#virt-attaching-vm-to-ovn-secondary-nw-cli_virt-connecting-vm-to-ovn-secondary-network
- Networking overview: https://79224--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/vm_networking/virt-networking-overview.html#secondary-network-config

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/virt-attaching-vm-secondary-network-cli.adoc [hunk 1]
```diff
@@ -29,20 +29,21 @@ spec:
...
-        - name: bridge-net <2>
+        - name: bridge-net <3>
           multus:
-            networkName: a-bridge-network <3>
+            networkName: a-bridge-network <4>
 ----
-<1> The name of the bridge interface.
-<2> The name of the network. This value must match the `name` value of the corresponding `spec.template.spec.domain.devices.interfaces` entry.
-<3> The name of the network attachment definition.
+<1> Optional: The name of the primary default pod network.
```
Comment (phoracek) [inline modules/virt-attaching-vm-secondary-network-cli.adoc:43]: Some people may understand this as - the `name` field is optional. While the whole items in the `interfaces` and `networks` lists is optional.

Vast majority of the customers attach to secondary networks only. I would suggest removing the default network from our "this is how you connect to a secondary network" examples altogether.

We could compensate for that removal in the network overview introduction:

> VMs can be connected to the default pod network and to any number of secondary networks.

Not sure about the exact wording, but it should explain that there can be 0/1 connections to the pod network and 0-N to secondary networks.

---

Alternatively we can adjust these callouts to refer to whole "default" items in `interfaces` and `networks`  sections.

File: virt/vm_networking/virt-networking-overview.adoc [hunk 2]
```diff
@@ -32,6 +32,8 @@ endif::openshift-rosa,openshift-dedicated[]
 [id="secondary-network-config"]
 == Configuring VM secondary network interfaces
 
+You can connect a virtual machine to a secondary network by using Linux bridge, SR-IOV and OVN-Kubernetes CNI plug-ins. You can list multiple secondary networks and interfaces in the VM specification. It is not required to specify the primary pod network in the VM specification when connecting to a secondary network interface.
```
Comment (phoracek) [inline virt/vm_networking/virt-networking-overview.adoc:35]: i like this

**Review by phoracek — APPROVED** [review]
LGTM

**Review by dfitzmau — COMMENTED** [review]
/remove-label peer-review-in-progress

/label peer-review-done

**Review by yossisegev — APPROVED** [review]
/lgtm

**Pull request merged**